src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
close.  this case was hit for any `ModelName` that looked like a top-level parameter or (after the fallback to empty prefix) property name.  e.g. the #1865 scenario where top-level POCO model was (due to this code) created the first time through.  so `CompositeModelBinder` didn't fallback to the empty prefix.